letsencrypt/acme/interfaces.py

```diff
@@ -17,6 +18,21 @@ def to_json():
             objects, that haven't been serialized yet, which is fine as
             long as :func:`letsencrypt.acme.util.dump_ijsonserializable`
             is used.
-        :rtype: dict
```

Comment:
That's because we want to use `cls.from_valid_json` classmethod (on class) and  `cls().to_json()` (on object, i.e. class instance), while `cls.to_json()` doesn't make much sense. Therefore a class definition that requires both must call `zope.interface.implements(IJSONSerializable)` and `zope.interface.classImplements(IJSONDeSerializable)` (note the difference btween `implements` and `classImplements`).

I'll put the above comment into `interfaces.__doc__`.